### PR TITLE
feat: Attach an image to a comment - MEED-2033 - Meeds-io/MIPs#50

### PR DESCRIPTION
**Pull request by SaraBoutej**
Add attach image to activity comment

### CODE REVIEW

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/comment/content/ActivityCommentRichText.vue [hunk 1]
```diff
@@ -101,11 +108,15 @@ export default {
       return !!this.commentId;
     },
     disableButton() {
-      return this.commenting || !this.message || !this.message.trim() || this.message.trim() === '<p></p>' || this.message.trim() === '<div></div>' || this.textLength > this.$activityConstants.COMMENT_MAX_LENGTH;
+      console.warn(this.activityCommentAttachmentsEdited);
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/comment/content/ActivityCommentRichText.vue]:
```suggestion
```

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue [hunk 2]
```diff
@@ -227,7 +227,6 @@ export default {
         }
         this.loading = true;
         this.$activityService.updateActivity(this.activityId, message, activityType, this.files, this.templateParams)
-          .then(this.postUpdateMessage)
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue]: you can keep this as is for future extensions. No need to delete it.

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 3]
```diff
@@ -9,7 +9,7 @@
       </div>
       <textarea
         ref="editor"
-        :id="ckEditorType"
+        :id="ckEditorID"
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]:
```suggestion
        :id="ckEditorInstanceId"
```

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 4]
```diff
@@ -121,6 +125,9 @@ export default {
     };
   },
   computed: {
+    ckEditorID() {
+      return this.ckEditorId ? this.ckEditorId : this.ckEditorType;
+    },
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]:
```suggestion
    ckEditorInstanceId() {
      return this.ckEditorId ? this.ckEditorId : this.ckEditorType;
    },
```

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 5]
```diff
@@ -212,7 +219,7 @@ export default {
     },
     initCKEditorInstance(reset, textValue) {
       this.inputVal = this.replaceWithSuggesterClass(textValue);
-      this.editor = CKEDITOR.instances[this.ckEditorType];
+      this.editor = CKEDITOR.instances[this.ckEditorID];
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]:
```suggestion
      this.editor = CKEDITOR.instances[this.ckEditorInstanceId];
```

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue [hunk 6]
```diff
@@ -294,7 +301,7 @@ export default {
         pasteFilter: 'p; a[!href]; strong; i', 
         on: {
           instanceReady: function () {
-            self.editor = CKEDITOR.instances[self.ckEditorType];
+            self.editor = CKEDITOR.instances[self.ckEditorID];
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]:
```suggestion
            self.editor = CKEDITOR.instances[self.ckEditorInstanceId];
```

Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue (hunk 5 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue (hunk 4 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue (hunk 3 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/common/components/RichEditor.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue (hunk 2 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue]: done

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/comment/content/ActivityCommentRichText.vue (hunk 1 above)
Comment (SaraBoutej) [reply webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/comment/content/ActivityCommentRichText.vue]: done

File: extension/war/src/main/webapp/WEB-INF/conf/social-extension/social/ckeditor-configuration.xml [hunk 7]
```diff
@@ -69,6 +69,24 @@
           </object>
         </object-param>
       </init-params>
+    </component-plugin>
+        <component-plugin>
```
Comment (boubaker) [inline extension/war/src/main/webapp/WEB-INF/conf/social-extension/social/ckeditor-configuration.xml]:
```suggestion
    <component-plugin>
```

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue [hunk 8]
```diff
@@ -255,7 +255,7 @@ export default {
               this.templateParams = activity.templateParams;
               return this.$nextTick().then(() => activity);
             })
-            .then(this.postSaveMessage)
+            .then(this.postUpdateMessage)
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/toolbar/ActivityComposerDrawer.vue]:
```suggestion
            .then(this.postSaveMessage)
```

File: webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/comment/list/ActivityComments.vue [hunk 9]
```diff
@@ -67,7 +67,7 @@ export default {
   },
   data: () => ({
     lastEditorOptions: null,
-    initializedComment: 0,
+    initializedComment: 0
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/activity-stream/components/comment/list/ActivityComments.vue]:
```suggestion
    initializedComment: 0,
```

File: webapp/portlet/src/main/webapp/vue-apps/attach-image/components/form/ImageInput.vue [hunk 10]
```diff
@@ -82,7 +82,6 @@ export default {
     triggerAttachmentsSave(event) {
       const objectType = event?.detail?.objectType;
       const objectId = event?.detail?.objectId;
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/attach-image/components/form/ImageInput.vue]:
```suggestion
      const objectId = event?.detail?.objectId;

```

File: webapp/portlet/src/main/webapp/vue-apps/attach-image/components/form/ImageInput.vue [hunk 11]
```diff
@@ -97,7 +97,6 @@ export default {
       const fileIds = this.attachments
         .filter(file => file.id)
         .map(file => file.id);
```
Comment (boubaker) [inline webapp/portlet/src/main/webapp/vue-apps/attach-image/components/form/ImageInput.vue]:
```suggestion
        .map(file => file.id);

```

**Review by boubaker — APPROVED** [review]

**Pull request merged**